core/tasks.py

```diff
@@ -238,3 +243,31 @@ def _overall_score_and_result_type(datafile):
     result_type = ScoreResult.ResultTypes.SIMPLE
     overall_score = float(datafile.readline())
     return overall_score, result_type
+
+
+def _get_mimetype(file_path):
+    mimetype = magic.from_file(file_path, mime=True)
+    # check if mimetype has something useful
+    # empty file returns inode/x-empty
+    if mimetype == 'inode/x-empty':
+        # if no mime type, it can be null(None)
+        return None
+    # Unsure about how to get file and content mime type seperately
+    mime_type = mimetype.split('/')
+    # file_mimetype = mime_type[0]
+    content_mimetype = mime_type[1]
+    """
+    check if the file is zipped since true
+    or false of uncompress has no effect on non-zip file
+    """
+    if content_mimetype == 'zip':
```

Comment:
There are many compression format that uncompress can handle (zip, gzip, bz2, probably others).  I would generically always check the `uncompress=True` type since we don't want to enumerate the compression formats that the magic library handles.

Reply:
Oh I did not take that into consideration. I will change it